king_phisher/client/dialogs/configuration.py

```diff
@@ -274,6 +291,27 @@ def interact(self):
 		self.dialog.destroy()
 		return response
 
+	def save_proxy_settings(self):
+		proxy_url = urllib.parse.urlparse(self.gtk_builder_get('entry_proxy_url').get_text().strip())
+		proxy_username = self.gtk_builder_get('entry_proxy_username').get_text().strip()
+		proxy_password = self.gtk_builder_get('entry_proxy_password').get_text().strip()
+		if proxy_url and not (proxy_url.hostname and proxy_url.scheme):
+			gui_utilities.show_dialog_warning('Invalid Proxy Settings', self.parent, 'The proxy url you have submitted is not valid.')
+			return
+		try:
+			proxy_url.port
+		except:
+			gui_utilities.show_dialog_warning('Invalid Port', self.parent, 'The port must be an integer.')
+			return
+		netloc = proxy_url.netloc
+		if proxy_username and proxy_password:
```

Comment:
Just check the `proxy_username` here to allow users to use proxies with users that don't have a password.